filter/chainsync/chainsync.go

```diff
@@ -206,9 +206,54 @@ func (c *ChainSync) Start() error {
 						for _, certificate := range v.Certificates {
 							switch certificate.(type) {
 							case *ledger.StakeDelegationCertificate:
```

Comment:
Make this `switch cert := certificate.(type) {` and you'll automatically have `cert` available as the correct type in each `case`

Reply:
Updated. Thanks...